Automattic/wp-calypso
# Change purpose of JetpackProductCard to be used for actual products

### Changes proposed in this Pull Request

`JetpackProductCard` used to be a generic card that didn't represent any real entity, and that was used as the base of the `JetpackPlanCard`, `JetpackBundleCard`, and `JetpackIndividualCard` components. What seemed to be a good idea at the time doesn't bring any benefit and seems to create confusion instead.

This PR:
- Removes `JetpackIndividualProductCard`
- Repurposes `JetpackProductCard` to represent an actual product (e.g. Jetpack Backup)

### Testing instructions

- In the dev docs (`/devdocs/design/jetpack-product-card`), check that `JetpackProductCard` represents an actual product (colour and copy).
- With the offer reset flow enabled (see screenshot), check that the _Plans_ page still looks the same (see screenshot).

### Screenshots

**Enable offer reset flow**
<img width="649" alt="Screen Shot 2020-08-10 at 2 06 13 PM" src="https://user-images.githubusercontent.com/1620183/89815562-013f2b00-db13-11ea-9d2c-107ca1ef420a.png">

**Plans page**
<img width="1144" alt="Screen Shot 2020-08-10 at 2 06 26 PM" src="https://user-images.githubusercontent.com/1620183/89815571-043a1b80-db13-11ea-9ca2-f7d243f951fc.png">
